nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -0,0 +1,349 @@
+                                                       .build()) {
+      for (GenericData.Record rec : records) {
+        writer.add(rec);
+      }
+    }
+
+    String location2 = table.location().replace("file:", "") + "/data/file2.avro";
+    try (FileAppender<GenericData.Record> writer = Avro.write(Files.localOutput(location2))
+                                                       .schema(schema)
+                                                       .named("test")
+                                                       .build()) {
+      for (GenericData.Record rec : records) {
+        writer.add(rec);
+      }
+    }
+
+    DataFile file1 = DataFiles.builder(table.spec())
+                              .withRecordCount(3)
+                              .withPath(location1)
+                              .withFileSizeInBytes(Files.localInput(location2).getLength())
+                              .build();
+
+    DataFile file2 = DataFiles.builder(table.spec())
+                              .withRecordCount(3)
+                              .withPath(location2)
+                              .withFileSizeInBytes(Files.localInput(location1).getLength())
+                              .build();
+
+    // add both data files
+    table.newAppend().appendFile(file1).appendFile(file2).commit();
+
+    // delete file2
+    table.newDelete().deleteFile(file2.path()).commit();
+
+    String manifestListLocation =
+        table.currentSnapshot().manifestListLocation().replace("file:", "");
+
+    List<ManifestFile> manifests = table.currentSnapshot().allManifests();
+
+    Assert.assertTrue(catalog.dropTable(TABLE_IDENTIFIER));
+    Assert.assertFalse(catalog.tableExists(TABLE_IDENTIFIER));
+
+    Assert.assertTrue(new File(location1).exists());
+    Assert.assertTrue(new File(location2).exists());
+    Assert.assertTrue(new File(manifestListLocation).exists());
+    for (ManifestFile manifest : manifests) {
+      Assert.assertTrue(new File(manifest.path().replace("file:", "")).exists());
+    }
+    Assert.assertTrue(new File(
+        ((HasTableOperations) table).operations()
+                                  .current()
+                                  .metadataFileLocation()
+                                  .replace("file:", ""))
+                             .exists());
+  }
+
+  @Test
+  public void testExistingTableUpdate() {
+    Table icebergTable = catalog.loadTable(TABLE_IDENTIFIER);
+    // add a column
+    icebergTable.updateSchema().addColumn("data", Types.LongType.get()).commit();
+
+    icebergTable = catalog.loadTable(TABLE_IDENTIFIER);
+
+    // Only 2 snapshotFile Should exist and no manifests should exist
+    Assert.assertEquals(2, metadataVersionFiles(TABLE_NAME).size());
+    Assert.assertEquals(0, manifestFiles(TABLE_NAME).size());
+    Assert.assertEquals(altered.asStruct(), icebergTable.schema().asStruct());
+
+  }
+
+  @Test(expected = CommitFailedException.class)
```

Comment:
Can you use `AssertHelpers.assertThrows` instead? That way, you can add assertions after the exception to check other things, like that the table has not been modified.

Reply:
:+1: